File: pos-scanner-web/backend/models/device_property.py
# backend/models/device_property.py
from extensions import db
from datetime import datetime


class DeviceProperty(db.Model):
    """设备性质模型 - 与商家ID关联"""
    __tablename__ = 'device_properties'

    id = db.Column(db.Integer, primary_key=True)
    merchant_id = db.Column(db.String(100), unique=True, nullable=False, index=True)
    property = db.Column(db.String(100), nullable=False)
    updated_at = db.Column(db.DateTime, default=datetime.now, onupdate=datetime.now)

    def to_dict(self):
        return {
            'merchantId': self.merchant_id,
            'property': self.property,
            'updatedAt': self.updated_at.isoformat() if self.updated_at else None
        }

File: pos-scanner-web/backend-go/internal/models/device_property.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type DeviceProperty struct {
	ID         uint           `gorm:"primaryKey" json:"id"`
	MerchantID string         `gorm:"size:100;uniqueIndex;not null" json:"merchant_id"`
	Property   string         `gorm:"size:100;not null" json:"property"`
	UpdatedAt  time.Time      `json:"updated_at"`
	DeletedAt  gorm.DeletedAt `gorm:"index" json:"-"`
}

func (DeviceProperty) TableName() string {
	return "device_properties"
}

func (d *DeviceProperty) ToDict() map[string]interface{} {
	return map[string]interface{}{
		"merchantId": d.MerchantID,
		"property":   d.Property,
		"updatedAt":  d.UpdatedAt.Format(time.RFC3339),
	}
}
